### PR TITLE
[Internal] Query: Adds flag for enabling string encoding in JsonBinaryWriter and default to true

### PR DESCRIPTION
**Pull request by neildsh**
## Description

Initial draft of change to add a flag for enabling/disabling string encoding in JsonBinaryWriter

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Json/JsonWriter.JsonBinaryWriter.cs [hunk 1]
```diff
@@ -31,7 +33,7 @@ abstract partial class JsonWriter : IJsonWriter
         /// <returns>Blitted bytes.</returns>
         public static PreblittedBinaryJsonScope CapturePreblittedBinaryJsonScope(Action<ITypedBinaryJsonWriter> scopeWriter)
         {
-            JsonBinaryWriter jsonBinaryWriter = new JsonBinaryWriter();
+            JsonBinaryWriter jsonBinaryWriter = new JsonBinaryWriter(initialCapacity: 256, serializeCount: false, enableEncodedStrings: true);
```
Comment (fnarenji) [inline Microsoft.Azure.Cosmos/src/Json/JsonWriter.JsonBinaryWriter.cs:36]: Set this to false here - preblitted scopes never want to have encoded strings enabled as they are context-dependent.
So far all preblitted scopes tacitly maintain that contract, but now that there a flag for it again we should set leverage it to have that property guaranteed by construction.

Comment (neildsh) [reply Microsoft.Azure.Cosmos/src/Json/JsonWriter.JsonBinaryWriter.cs]: Changed this

File: Microsoft.Azure.Cosmos/src/Json/JsonWriter.JsonBinaryWriter.cs [hunk 2]
```diff
@@ -827,20 +832,23 @@ private void WriteFieldNameOrString(bool isFieldName, Utf8Span utf8Span)
...
-                else if (isFieldName
+                else if (this.EnableEncodedStrings
+                    && isFieldName
                     && (utf8Span.Length >= MinReferenceStringLength)
                     && this.TryRegisterStringValue(utf8Span))
                 {
                     // Work is done in the check
                 }
-                else if (!isFieldName
+                else if (this.EnableEncodedStrings
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/Json/JsonWriter.JsonBinaryWriter.cs:842]: >EnableEncodedStrings [](http://example.com/codeflow?start=30&length=20)

this basically expands the flag to include all encoded string (reference + compressed). Is this intentinal?

Comment (neildsh) [reply Microsoft.Azure.Cosmos/src/Json/JsonWriter.JsonBinaryWriter.cs:842]: yes, it's intentional

**Review by sboshra — APPROVED** [review]
:shipit:

**Review by sboshra — APPROVED** [review]

**Pull request merged**